src/vm/variables.rs

```diff
@@ -19,20 +19,41 @@ use vm::contexts::{Environment, LocalContext};
 use vm::errors::{InterpreterResult as Result, RuntimeErrorType};
 use vm::types::Value;
 use vm::types::{BuffData, OptionalData, PrincipalData};
+use vm::ClarityVersion;
 
 use vm::costs::cost_functions::ClarityCostFunction;
 use vm::costs::runtime_cost;
 
-define_named_enum!(NativeVariables {
-    ContractCaller("contract-caller"), TxSender("tx-sender"), BlockHeight("block-height"),
-    BurnBlockHeight("burn-block-height"), NativeNone("none"),
-    NativeTrue("true"), NativeFalse("false"),
-    TotalLiquidMicroSTX("stx-liquid-supply"),
-    Regtest("is-in-regtest"), TxSponsor("tx-sponsor?"),
+define_versioned_named_enum!(NativeVariables(ClarityVersion) {
+    ContractCaller("contract-caller", ClarityVersion::Clarity1),
+    TxSender("tx-sender", ClarityVersion::Clarity1),
+    BlockHeight("block-height", ClarityVersion::Clarity1),
+    BurnBlockHeight("burn-block-height", ClarityVersion::Clarity1),
+    NativeNone("none", ClarityVersion::Clarity1),
+    NativeTrue("true", ClarityVersion::Clarity1),
+    NativeFalse("false", ClarityVersion::Clarity1),
+    TotalLiquidMicroSTX("stx-liquid-supply", ClarityVersion::Clarity1),
+    Regtest("is-in-regtest", ClarityVersion::Clarity1),
+    TxSponsor("tx-sponsor?", ClarityVersion::Clarity2),
```

Comment:
So, we don't have a coding convention anywhere, but this is the first time I'm seeing `?` getting used in a variable name.  Is this a convention we want?

Reply:
I think it means that the return value is "(option X)"?

https://github.com/blockstack/stacks-blockchain/blob/2c424d9142a1b8a8ad53ceca6c4c96a33bc201a1/src/vm/variables.rs#L59